kong/admin_service_test.go

```diff
@@ -0,0 +1,180 @@
+// +build enterprise
+
+package kong
+
+import (
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAdminService(T *testing.T) {
+	runWhenEnterprise(T, ">=0.33.0", false)
+	assert := assert.New(T)
+
+	client, err := NewTestClient(nil, nil)
+	assert.Nil(err)
+	assert.NotNil(client)
+
+	admin := &Admin{
+		Email:            String("admin@test.com"),
+		Username:         String("newAdmin"),
+		CustomID:         String("admin123"),
+		RBACTokenEnabled: Bool(true),
+	}
+
+	createdAdmin, err := client.Admins.Create(defaultCtx, admin)
+	assert.Nil(err)
+	assert.NotNil(createdAdmin)
+
+	admin, err = client.Admins.Get(defaultCtx, createdAdmin.ID, false)
+	assert.Nil(err)
+	assert.NotNil(admin)
+
+	admin.CustomID = String("admin321")
+	admin, err = client.Admins.Update(defaultCtx, admin)
+	assert.Nil(err)
+	assert.NotNil(admin)
+	assert.Equal("admin321", *admin.CustomID)
+
+	err = client.Admins.Delete(defaultCtx, createdAdmin.ID)
+	assert.Nil(err)
+}
+
+func TestAdminServiceWorkspace(T *testing.T) {
+	runWhenEnterprise(T, ">=0.33.0", false)
```

Comment:
These tests with `rbac=false` won't be executed in the current CI pipeline. Does this mean that we need to include another matrix parameter in the CI to run all tests with combinations of rbac=true and false?

What is kong's behavior here if you run this test with rbac=off in kong?

Reply:
Kong will throw 500s if a request is made to an endpoint that utilizes RBAC entities with `rbac=off`. `TestAdminServiceRegisterCredentials` is an example of this.

I believe these tests will run in the current pipeline. When the `rbac` parameter in `runWhenEnterprise` is true it signals that this test needs RBAC enabled to run properly. I think that the EE tests in the CI pipeline should be run with RBAC enabled by default. I don't see a downside to that. I included this additional parameter in case folks needed to run the test suite against an instance of Kong EE that for whatever reason didn't have RBAC enabled. 

I'm going to rename the function parameter for more clarity